src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java

```diff
@@ -313,6 +317,25 @@ public Object convert(Class type, Object value) {
 
     }
 
+    /** A converter that converts a string array to a pattern. */
+    private static final class PatternArrayConverter implements Converter {
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public Object convert(Class type, Object value) {
+            final String plainString = value.toString();
+            Pattern[] patterns = new Pattern[0];
+            if (!CommonUtil.isBlank(plainString)) {
+                final String[] patternStrings = plainString.split(REGEX_SEPARATOR);
+                patterns = new Pattern[patternStrings.length];
+                for (int index = 0; index < patternStrings.length; index++) {
+                    patterns[index] = CommonUtil.createPattern(patternStrings[index]);
```

Comment:
Do we need index variable here ? Can we loop in form of for-each.

Reply:
done, replaced with for-each